src/App.jsx

```diff
@@ -14,33 +16,89 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const ALPHABET = 'alphabet';
+const LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+const prepareGoods = (sortMethod, reverseList) => {
+  const copyGoods = [...goodsFromServer];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortMethod) {
+    copyGoods.sort((a, b) => {
+      if (sortMethod === ALPHABET) {
+        return a.localeCompare(b);
+      }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+      if (sortMethod === LENGTH) {
+        return a.length - b.length;
+      }
+
+      return 0;
+    });
+  }
+
+  if (reverseList) {
+    copyGoods.reverse();
+  }
+
+  return copyGoods;
+};
+
+export const App = () => {
+  const [sortMethod, setSortMethod] = useState('');
+  const [reverseList, setReverseList] = useState(false);
+
+  const resetAll = () => {
+    setSortMethod('');
+    setReverseList(false);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const goodsList = prepareGoods(sortMethod, reverseList);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortMethod(ALPHABET)}
+          type="button"
+          className={`button is-info ${sortMethod !== ALPHABET && 'is-light'}`}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => setSortMethod(LENGTH)}
+          type="button"
+          className={`button is-success ${sortMethod !== LENGTH && 'is-light'}`}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => setReverseList(!reverseList)}
+          type="button"
+          className={`button is-warning ${!reverseList && 'is-light'}`}
+        >
+          Reverse
+        </button>
+
+        {(sortMethod || reverseList) && (
```

Comment:
Create a variable for `sortMethod || reverseList` to avoid combining different logical operators